libs/api_common/device_helper.cpp

```diff
@@ -27,21 +27,35 @@ typedef struct _async_ioctl_completion_context
 } async_ioctl_completion_context_t;
 
 // Handle used for synchronous calls to the driver.
-static ebpf_handle_t _sync_device_handle = ebpf_handle_invalid;
+static thread_local ebpf_handle_t _sync_device_handle = ebpf_handle_invalid;
```

Comment:
Would it make sense to implement `_sync_device_handle` and `_async_device_handle` as instances of a new `device_handle` class (w/constr(), destr() and get())?

Reply:
I tried this approach at first, but this causes ab order of operations issue with the test code. We need to have the handles closed before the global destructors are called for the CloseHandle mocks in the test code. DLL detach is called before global destructor, so the test code works correctly.

If we can figure out how to sequence thread_local destructor with destruction of static globals, then maybe we could make this work.

Until then, this is the simplest approach that works.